### ISSUE TITLE
PluginBase causes ImportError in PyYAML

### ISSUE THREAD

**Issue by cesarvandevelde** [mitsuhiko/pluginbase#5]
PluginBase seems to break PyYAML, causing ImportErrors on import.  
I'm using the latest version of PyYAML (3.11) and PluginBase.  
Tested on OS X and Linux.

Importing PluginBase first causes errors:
```
Python 2.7.3 (default, Mar 18 2014, 05:13:23) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pluginbase
>>> import yaml
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/pluginbase.py", line 404, in plugin_import
    fromlist, level)
  File "/usr/local/lib/python2.7/dist-packages/yaml/__init__.py", line 2, in <module>
    from error import *
  File "/usr/local/lib/python2.7/dist-packages/pluginbase.py", line 404, in plugin_import
    fromlist, level)
ImportError: No module named error
```

Importing YAML first works fine:
```
Python 2.7.3 (default, Mar 18 2014, 05:13:23) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import yaml
>>> import pluginbase
```

**Comment by akun** [comment]
also causes ImportError in bsddb

```
Python 2.6.5 (r265:79063, Feb 27 2014, 19:43:51)
[GCC 4.4.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pluginbase
>>> import bsddb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/data/users/zhengk/release/lib/python2.6/site-packages/pluginbase.py", line 404, in plugin_import
    fromlist, level)
  File "/usr/lib64/python2.6/bsddb/__init__.py", line 65, in <module>
    from bsddb.dbutils import DeadlockWrap as _DeadlockWrap
  File "/var/data/users/zhengk/release/lib/python2.6/site-packages/pluginbase.py", line 404, in plugin_import
    fromlist, level)
  File "/usr/lib64/python2.6/bsddb/dbutils.py", line 35, in <module>
    import db
  File "/var/data/users/zhengk/release/lib/python2.6/site-packages/pluginbase.py", line 404, in plugin_import
    fromlist, level)
ImportError: No module named db
```

Importing bsddb first works fine:

```
Python 2.6.5 (r265:79063, Feb 27 2014, 19:43:51)
[GCC 4.4.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import bsddb
>>> import pluginbase
```

**Comment by nfarrar** [comment]
Yeah, won't work with PyYAML. :(

**Comment by mittagessen** [comment]
I am encountering the same issue and it seems to be related to relative imports in the "offending" packages while the importer assumes absolute imports. I've been able to fix the issue by changing the last few relative imports in the "offending" package. I'll look at the pluginbase source code in the next few days and hopefully create a pull request.

**Comment by darvid** [comment]
any updates on this? there are a number of libraries in the stdlib that are affected by this as well, including ``xml.sax.saxutils``

**Comment by mittagessen** [comment]
I am sorry but I've switched to [stevedore](http://docs.openstack.org/developer/stevedore) as it seems to be in active development and integrates nicely into setuptools.

**Comment by akun** [comment]
@mittagessen Thanks!

**Comment by PaulSec** [comment]
Same issue here with Scapy.

**Comment by phretor** [comment]
All same issues. See also #10 

Is this module in active development?

**Comment by PaulSec** [comment]
It does not seem too. I re-implemented my own Plugin system to be honest.

**Comment by zeroSteiner** [comment]
I am also experiencing this issue.

**Comment by PaulSec** [comment]
In case you're interested, I create a small repo to show how I handled a minimalistic Plugin architecture - https://github.com/PaulSec/Python-plugins

**Comment by zeroSteiner** [comment]
I was able to fix the issue by changing `_ImportHook.plugin_import(... level=0)` to `...level=-1)` which according to the [docs](https://docs.python.org/2/library/functions.html#__import__) will allow for both relative and absolute imports.

**Comment by zeroSteiner** [comment]
I fixed this in commit 9e5764f1316ffc8819dd221153d87158a33ec895 by setting the level parameter to the default value for the version of Python which is running. This appears to have been an issue with level always defaulting to 0 in Python 2 which prevents relative imports from working. See the documentation of [\__import__](https://docs.python.org/2/library/functions.html#__import__) for more details.
